### DISCUSSION TITLE
Error retrieving credentials from the instance profile metadata service.

### DISCUSSION THREAD

**Discussion by alexfleites** [aws/aws-sdk-php-laravel#219] [Q&A]
Some body have experience this issue before, working win docker using lando.dev getting this error and cannot find a solution: "Error retrieving credentials from the instance profile metadata service. (cURL error 7: Failed to connect to 169.254.169.254 port 80: Connection refused (see https://curl.haxx.se/libcurl/c/libcurl-errors.html))"

BTW aws credentials are already created and added to the .env file.

**Comment by yenfryherrerafeliz** [comment]
Hi @alexfleites, thanks for opening this discussion. Which type of credential provider are you trying to use?, if you want to use credential from the environment you could specify it when instancing the client. Example below:

```php

$envProvider = CredentialProvider::env();
$client = new S3Client([
    'version' => 'latest',
    'region' => 'REGION',
    'credentials' => $envProvider
]);
``` 

If this does not work, so then you need to make sure the variables are being set.

Thanks!
